README.md

```diff
@@ -1,88 +1,167 @@
-  // Cloud monitor messages fired to cb
-});
+## How to use
+The MQTT connection over Websocket relies on Username / Password authentication. Under the hood, this module uses your user ID (plus a timestamp) as *Username* and needs a valid JWT Token as *Password*. You can use either a valid JWT token or just your API Credentials (*clientId* and *clientSecret*).
 
-ArduinoCloud.writeCloudMonitor(deviceId, message).then(() => {
-  // Message sent to cloud monitor
-});
+### How to import arduino-iot-js in your project
+Using a web application in the browser
+```javascript
+import ArduinoIoTCloud from 'arduino-iot-js'
+```
+Using nodejs
+```javascript
+const { ArduinoIoTCloud } = require('arduino-iot-js');
+```
 
-ArduinoCloud.closeCloudMonitor(deviceId).then(topic => {
-  // Close cloud monitor
+### How to connect to Arduino IoT Cloud broker using API Credentials
+```javascript
+const { ArduinoIoTCloud } = require('arduino-iot-js');
+
+const options = {
+    clientId: "YOUR_CLIENT_ID",
+    clientSecret: "YOUR_CLIENT_SECRET",
+    onDisconnect: message => {
+        console.error(message);
+    }
+}
+
+ArduinoIoTCloud.connect(options).then(() => {
+    console.log("Connected to Arduino IoT Cloud broker");
+}, error => {
+    console.error(error);
 });
+```
 
-// Send a property value to a device
-// - value can be a string, a boolean or a number
-// - timestamp is a unix timestamp, not required
-ArduinoCloud.sendProperty(thingId, name, value, timestamp).then(() => {
-  // Property value sent
-});
+### How to listen for property value updates
+After a successful connection, you can listen for property updates.
+To do this you need:
+- The ID of the *Thing* the *property* belongs to. You can list all your things and properties using the [Javascript Rest API client](https://www.npmjs.com/package/@arduino/arduino-iot-client), calling the [GET Things endpoint](https://www.arduino.cc/reference/en/iot/api/index.html#api-ThingsV2-thingsV2List)
+- The *variable name* of the property you want to listen
 
-// Register a callback on a property value change
-// 
-ArduinoCloud.onPropertyValue(thingId, propertyName, updateCb).then(() => {
-  // updateCb(message) will be called every time a new value is available. Value can be string, number, or a boolean depending on the property type
+```javascript
+const { ArduinoIoTCloud } = require('arduino-iot-js');
+const thingID = "THING_ID"
+const variableName = "PROPERTY_NAME"
+
+const options = {
+    clientId: "YOUR_CLIENT_ID",
+    clientSecret: "YOUR_CLIENT_SECRET",
+    onDisconnect: message => {
+        console.error(message);
+    }
+}
+
+ArduinoIoTCloud.connect(options).then(() => {
+    console.log("Connected to Arduino IoT Cloud broker");
+    ArduinoIoTCloud.onPropertyValue(thingId, variableName, showUpdates = value => {
+        console.log(value);
+    }).then(() => {
+        console.log("Callback registered");
+    }, error => {
+        console.log(error);
+    });
 });
+```
```

Comment:
```suggestion
ArduinoIoTCloud.connect(options)
  .then(() => {
    console.log("Connected to Arduino IoT Cloud broker");
    return ArduinoIoTCloud.onPropertyValue(thingId, variableName,
      value => console.log(value));
  })
  .then(() => console.log("Callback registered"))
  .catch(error => console.log(error));
```

Reply:
I think that using an anonymous function as property update callback doesn't work